File: api/JsonPlaceholderApi.js
import BaseApi from "./BaseApi.js"
import testData from "../utils/testData.js"
import headerData from "../utils/headerData.js"

class JsonPlaceholderApi extends BaseApi {

    static async getAllPosts(){
        return await BaseApi.getRequest(testData.url, testData.request.posts)
    }

    static async get99Posts(){
        return await BaseApi.getRequest(testData.url, testData.request.posts99)
    }

    static async get150Posts(){
        return await BaseApi.getRequest(testData.url, testData.request.posts150)
    }

    static async postRandomText(dataRandom){
        const data = JSON.stringify(dataRandom)
        return await BaseApi.postRequest(testData.url, testData.request.posts, headerData.headerJson, data)
    }

    static async getUsers(){
        return await BaseApi.getRequest(testData.url, testData.request.users)
    }

    static async get5Users(){
        return await BaseApi.getRequest(testData.url, testData.request.users5)
    }

}
export default JsonPlaceholderApi

File: Api/JsonplaceholderApi.py
from AutomationQATrainee.Api.BaseApi import BaseApi
from AutomationQATrainee.Utils.ConfigUtils import ConfigUtils
from AutomationQATrainee.Utils.GetTestData import GetTestData
import json


class JsonplaceholderApi(BaseApi):

    @staticmethod
    def get_all_posts():
        return BaseApi.get_request(ConfigUtils.get_utils_from_json('url'),
                                   GetTestData.get_data_from_json('request', 'posts'))

    @staticmethod
    def get_99post():
        return BaseApi.get_request(ConfigUtils.get_utils_from_json('url'),
                                   GetTestData.get_data_from_json('request', 'posts99'))

    @staticmethod
    def get_150post():
        return BaseApi.get_request(ConfigUtils.get_utils_from_json('url'),
                                   GetTestData.get_data_from_json('request', 'posts150'))

    @staticmethod
    def post_random_text(random_data):
        data = json.dumps(random_data)
        return BaseApi.post_request(ConfigUtils.get_utils_from_json('url'),
                                    GetTestData.get_data_from_json('request', 'posts'),
                                    data, ConfigUtils.get_utils_from_json('headerJson'))
